slrealizer/realize_om10.py

```diff
@@ -0,0 +1,298 @@
+
+        ####################################
+        # Merging catalog with observation #
+        ####################################
+        
+        observation = self.observation.copy()
+        catalog['key'] = 0
+        observation['key'] = 0
+        src = catalog.merge(observation, how='left', on='key')
+        src.drop('key', 1, inplace=True)
+        gc.collect()
+        
+        ####################################
+        # Observation parameter conversion #
+        ####################################
+        
+        src['apFluxErr'] = from_mag_to_flux(src['fiveSigmaDepth']-22.5)/5.0 # because Fb = 5 \sigma_b
+        src['sigmaSqPSF'] = np.power(fwhm_to_sigma(src['FWHMeff']), 2.0)
+        # Arbitrarily set REFF_T to 1.0
+        #src['sigmaSqLens'] = np.power(hlr_to_sigma(src['REFF_T']), 2.0)
+        src['sigmaSqLens'] = np.power(hlr_to_sigma(1.0), 2.0)
+        src.rename(columns={'FWHMeff': 'psf_fwhm'}, inplace=True) 
+        src.drop(['fiveSigmaDepth', 'REFF_T'], axis=1, inplace=True)
+        gc.collect()
+        
+        ###############################
+        # Adaptive moment calculation #
+        ###############################
+        
+        # Set unused band magnitudes to zero, and
+        # work with one magnitude in the observed filter
+        for b in 'ugriz': # b = observed filter
+            setZeroLens = lensMagCols[:]
+            setZeroLens.remove(b + '_SDSS_lens')
+            setZeroQ = qMagCols[:]
+            setZeroQ.remove(b + '_SDSS_quasar')
+            src.loc[src['filter'] == b, setZeroLens] = 0.0
+            src.loc[src['filter'] == b, setZeroQ] = 0.0
+        src['lens_mag'] = src[lensMagCols].sum(axis=1)
+        src['q_mag'] = src[qMagCols].sum(axis=1)
+        src.drop(lensMagCols + qMagCols, axis=1, inplace=True)
+        gc.collect()
+        
+        # Convert magnitudes into fluxes
+        src['lens_flux'] = from_mag_to_flux(src['lens_mag'], to_unit='nMgy')
+        for q in range(4):
+            src['q_flux_' + str(q)] = from_mag_to_flux(src['q_mag'] +\
+                                                       from_flux_to_mag(np.abs(src['MAG_' + str(q)])),\
+                                                       to_unit='nMgy')
+
+        # Set fluxes of nonexistent quasar images to zero
+        src.loc[src['NIMG'] == 2, ['q_flux_2', 'q_flux_3']] = 0.0
+        src.loc[src['NIMG'] == 3, ['q_flux_3']] = 0.0
+            
+        # Get total flux
+        fluxCols = ['q_flux_' + str(q) for q in range(4)] + ['lens_flux']
+        src['apFlux'] = src[fluxCols].sum(axis=1)
+
+        # Calculate flux ratios (for moment calculation)
+        src['lensFluxRatio'] = src['lens_flux']/src['apFlux']
+        for q in range(4):
+            src['qFluxRatio_' + str(q)] = src['q_flux_' + str(q)]/src['apFlux']
+        src.drop(['lens_flux'] + ['q_flux_' + str(q) for q in range(4)], axis=1, inplace=True)
+        gc.collect()
+
+        # FIRST MOMENTS
```

Comment:
Is all this analytic moment code a duplication of the analytic moments code in the `_om10_to_lsst` method, above? Is it possible to write a utility method that is called by both, and takes either scalar or vector arguments?

Reply:
Refactored to use the `_include_moments` method in the SLRealizer parent class! One of the unit tests in `tests/test_om10realizer.py` checks if this was done right.